noxfile.py

```diff
@@ -52,9 +54,43 @@ def benchmark_model(session: nox.Session, subpackage: ModelSubpackagePath) -> No
     """Run pytest benchmarks for selected icon4py model subpackages."""
     _install_session_venv(session, extras=["dace", "io", "testing"], groups=["test"])
 
+    results_json_path = os.path.abspath(f"results_{subpackage.replace('/', '_')}.json")
     with session.chdir(f"model/{subpackage}"):
-        session.run(*"pytest -sv --benchmark-only".split(), *session.posargs)
-
+        session.run(*f"pytest \
+                       -v \
+                       --benchmark-only \
+                       --benchmark-warmup=on \
+                       --benchmark-warmup-iterations=30 \
+                       --benchmark-json={results_json_path}".split(),
+                       *session.posargs)
+
+@nox.session(python=["3.10", "3.11"], requires=["benchmark_model-{python}" + f"({subpackage.id})" for subpackage in MODEL_SUBPACKAGE_PATHS])
```

Comment:
Have you tried to require the session without any argument? At least from the command line, when you omit the arguments it will run that session for all possible arguments so I guess it should work here also.
```suggestion
@nox.session(python=["3.10", "3.11"], requires=["benchmark_model-{python}"])
```

Reply:
That was the first thing I tried, but it does not work :)
It runs only for one of the sub packages.